### PR TITLE
Simplified scripts for copying to jb-main

### PR DESCRIPTION
**Pull request by igordmn**
Now we need only to switch to the commit, and call a needed script. It creates 2 branches, which can be tested on CI, and merged with simple git merge.

Branches:
image

### CODE REVIEW

File: scripts/snap/impl/snapToJbMain.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+set -e
+
+if [ -z "$1" ]; then
+echo "Specify the snapping subfolder. For example: ./snapToJbMain.sh androidx/compose-ui/1.6.0-alpha02 compose ':(exclude)compose/material3'"
```
Comment (MatkovIvan) [inline scripts/snap/impl/snapToJbMain.sh]: Description for args is not clear. What is branch? Out? In? Is it possible to specify hash? What is folder? What should be current git branch?

File: scripts/snap/snapNavigation.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+$(dirname "$0")/impl/snapToJbMain.sh 'compose/navigation'
```
Comment (MatkovIvan) [inline scripts/snap/snapNavigation.sh]:
```suggestion
$(dirname "$0")/impl/snapToJbMain.sh 'navigation'
```

File: scripts/snap/snapSavedstate.sh [hunk 3]
```diff
@@ -0,0 +1 @@
+$(dirname "$0")/impl/snapToJbMain.sh 'compose/savedstate'
```
Comment (MatkovIvan) [inline scripts/snap/snapSavedstate.sh]:
```suggestion
$(dirname "$0")/impl/snapToJbMain.sh 'savedstate'
```

File: scripts/snap/snapLifecycle.sh [hunk 4]
```diff
@@ -0,0 +1 @@
+$(dirname "$0")/impl/snapToJbMain.sh 'compose/lifecycle'
```
Comment (MatkovIvan) [inline scripts/snap/snapLifecycle.sh]:
```suggestion
$(dirname "$0")/impl/snapToJbMain.sh 'lifecycle'
```

File: scripts/snap/snapAnnotation.sh [hunk 5]
```diff
@@ -0,0 +1 @@
+$(dirname "$0")/impl/snapToJbMain.sh 'compose/annotation'
```
Comment (MatkovIvan) [inline scripts/snap/snapAnnotation.sh]:
```suggestion
$(dirname "$0")/impl/snapToJbMain.sh 'annotation'
```

File: scripts/snap/snapCollection.sh [hunk 6]
```diff
@@ -0,0 +1 @@
+$(dirname "$0")/impl/snapToJbMain.sh 'compose/collection'
```
Comment (MatkovIvan) [inline scripts/snap/snapCollection.sh]:
```suggestion
$(dirname "$0")/impl/snapToJbMain.sh 'collection'
```

File: scripts/snap/impl/snapToJbMain.sh [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+fi
+
+DIR=$(dirname "$0")
+ALL_FOLDERS=${@:1}
+CURRENT_COMMIT=$(git rev-parse --short @)
+BRANCH_TO_RESTORE_IN_THE_END=$(git branch --show-current)
+
+
+TO_JB_MAIN_BRANCH=integration-snap/$CURRENT_COMMIT/to-jb-main
+git checkout --quiet $(git merge-base $CURRENT_COMMIT origin/jb-main) -B $TO_JB_MAIN_BRANCH
```
Comment (MatkovIvan) [inline scripts/snap/impl/snapToJbMain.sh]: Doesn't work for my case: remote can have not "origin" name

File: scripts/snap/impl/snapToJbMain.sh (hunk 1 above)
Comment (igordmn) [reply scripts/snap/impl/snapToJbMain.sh]: This shouldn't be used explicitly. Added a comment to look the parent scripts/readme

File: scripts/snap/impl/snapToJbMain.sh (hunk 7 above)
Comment (igordmn) [reply scripts/snap/impl/snapToJbMain.sh]: removed. local branches should be up-to-date to this point anyway.

Comment (igordmn) [reply scripts/snap/impl/snapToJbMain.sh]: > local branches should be up-to-date to this point anyway.

Actually, I am wrong, all work can be done in "integration" without updating "jb-main".

Fixed reading the associated remote.

**Review by MatkovIvan — APPROVED** [review]

**Comment by igordmn** [comment]
Let's not merge it, as we are [discussing](https://jetbrains.slack.com/archives/G017NLN12D8/p1722633195837899) to not use "snaps" anymore.

**Comment by igordmn** [comment]
These scripts still simplify process. I changed them for "copying" approach

File: scripts/copy/Readme.txt [hunk 8]
```diff
@@ -0,0 +1,11 @@
+The purposes of these scripts is to copy some subfolder to jb-main branch from "integration" or "integration-release/*".
+
+1. Checkout the commit you want to copy ("integration" or "integration-release/*")
+
+2. Merge jb-main to integration, pick "jb-main" state in a case of conflicts in other folders
+
+3. Call ./copyCompose.sh (use another script for another folder)
```
Comment (MatkovIvan) [inline scripts/copy/Readme.txt]:
```suggestion
3. Call `./copyCompose.sh` (use another script for another folder)
```

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**